web3/utils/module_testing/eth_module.py

```diff
@@ -442,6 +443,80 @@ def test_eth_newPendingTransactionFilter(self, web3):
         result = web3.eth.uninstallFilter(filter.filter_id)
         assert result is True
 
+    def test_eth_getLogs_block_range_no_logs(self, web3, block_with_txn_with_log):
+        # Test with block range
+
+        filter_params = {
+            "fromBlock": 0,
+            "toBlock": block_with_txn_with_log['number'] - 1,
+        }
+        result = web3.eth.getLogs(filter_params)
+        assert len(result) == 0
+
+        # the range is wrong
+        filter_params = {
+            "fromBlock": block_with_txn_with_log['number'],
+            "toBlock": block_with_txn_with_log['number'] - 1,
+        }
+        result = web3.eth.getLogs(filter_params)
+        assert len(result) == 0
+
+        # Test with `address`
+
+        # filter with other address
+        filter_params = {
+            "fromBlock": 0,
+            "address": UNKNOWN_ADDRESS,
+        }
+        result = web3.eth.getLogs(filter_params)
+        assert len(result) == 0
+
+    def test_eth_getLogs_block_range_with_logs(self,
+                                               web3,
+                                               block_with_txn_with_log,
+                                               emitter_contract,
+                                               txn_hash_with_log):
+        def assert_is_emitted_log(log_entry):
+            assert log_entry['blockNumber'] == block_with_txn_with_log['number']
+            assert log_entry['blockHash'] == block_with_txn_with_log['hash']
+            assert log_entry['logIndex'] == 0
+            assert is_same_address(log_entry['address'], emitter_contract.address)
+            assert log_entry['transactionIndex'] == 0
+            assert log_entry['transactionHash'] == HexBytes(txn_hash_with_log)
+
+        # Test with block range
+
+        # the range includes the block where the log resides in
+        filter_params = {
+            "fromBlock": block_with_txn_with_log['number'],
+            "toBlock": block_with_txn_with_log['number'],
+        }
+        result = web3.eth.getLogs(filter_params)
+        assert len(result) == 1
+        log_entry = result[0]
+        assert_is_emitted_log(log_entry)
```

Comment:
Totally at your discretion: this _might_ be a bit cleaner if the assertion did the length check and log extraction. (something like `assert_contains_log(result)`)

Reply:
Have changed it. Thank you for the suggestion!